multichain-api-gateway/src/lib.rs

```diff
@@ -0,0 +1,201 @@
+use std::{net::TcpListener, str};
+
+use actix_web::{dev::Server, web, App, HttpRequest, HttpResponse, HttpServer};
+use futures::{stream, StreamExt};
+use reqwest::Client;
+use serde_json::Value;
+use url::Url;
+
+use crate::config::{BlockScoutSettings, Instance, Settings};
+
+pub mod config;
+
+fn build_urls(path: &str, query: &str, settings: &BlockScoutSettings) -> Vec<(Instance, Url)> {
+    settings
+        .instances
+        .iter()
+        .map(|Instance(net, subnet)| {
+            let mut url = settings.base_url.clone();
+            url.set_path(&format!("{}/{}{}", net, subnet, path));
+            url.set_query(Some(query));
+            (Instance(net.clone(), subnet.clone()), url)
+        })
+        .collect::<Vec<_>>()
+}
+
+async fn make_requests(
+    urls: Vec<(Instance, Url)>,
+    concurrent_requests: usize,
+) -> Vec<(Instance, String)> {
+    let client = Client::new();
+
+    stream::iter(urls)
+        .map(|(instance, url)| {
+            let client = &client;
+            async move {
+                let resp = client.get(url).send().await.unwrap();
+                (instance.clone(), resp.bytes().await)
```

Comment:
```suggestion
            async {
                let resp = client.get(url).send().await.unwrap();
                (instance.clone(), resp.bytes().await)
```
You may do it the following way without copying the reference